mods/d2k/maps/atreides-05/atreides05.lua

```diff
@@ -0,0 +1,415 @@
+		{ "trike", "trike", "quad", "quad", "quad", "trike", "trike", "trike", "trike", "trike", "trike", "quad", "quad" },
+		{ "combat_tank_o", "combat_tank_o", "combat_tank_o", "combat_tank_o", "combat_tank_o", "combat_tank_o", "trike", "trike", "quad", "quad", "light_inf", "light_inf", "light_inf", "trooper", "trooper", "trooper", "trooper" }
+	}
+}
+
+MercenaryAttackDelay =
+{
+	easy = DateTime.Minutes(2) + DateTime.Seconds(40),
+	normal = DateTime.Minutes(1) + DateTime.Seconds(50),
+	hard = DateTime.Minutes(1) + DateTime.Seconds(10)
+}
+
+MercenaryAttackWaves =
+{
+	easy = 3,
+	normal = 6,
+	hard = 9
+}
+
+MercenarySpawn = { HarkonnenRally4.Location + CVec.New(2, -2) }
+
+-- Ordos tanks because those were intended for the Smugglers not the Atreides
+ContrabandReinforcements = { "mcv", "quad", "quad", "combat_tank_o", "combat_tank_o", "combat_tank_o" }
+SmugglerReinforcements = { "quad", "quad", "trike", "trike" }
+InitialHarkonnenReinforcements = { "trooper", "trooper", "quad", "quad", "trike", "trike", "trike", "light_inf" }
+
+HarkonnenPaths =
+{
+	{ HarkonnenEntry1.Location, HarkonnenRally1.Location },
+	{ HarkonnenEntry1.Location, HarkonnenRally3.Location },
+	{ HarkonnenEntry1.Location, HarkonnenRally2.Location },
+	{ HarkonnenEntry1.Location, HarkonnenRally4.Location },
+	{ HarkonnenEntry2.Location }
+}
+
+AtreidesReinforcements = { "trike", "combat_tank_a" }
+AtreidesPath = { AtreidesEntry.Location, AtreidesRally.Location }
+
+ContrabandTimes =
+{
+	easy = DateTime.Minutes(10),
+	normal = DateTime.Minutes(5),
+	hard = DateTime.Minutes(2) + DateTime.Seconds(30)
+}
+
+wave = 0
+SendHarkonnen = function()
+	Trigger.AfterDelay(HarkonnenAttackDelay[Difficulty], function()
+		if player.IsObjectiveCompleted(KillHarkonnen) then
+			return
+		end
+
+		wave = wave + 1
+
+		if InfantryReinforcements and wave % 4 == 0 then
+			local inf = Reinforcements.Reinforce(harkonnen, HarkonnenInfantryReinforcements[Difficulty][wave/4], InfantryPath)
+			Utils.Do(inf, function(unit)
+				unit.AttackMove(HarkonnenAttackLocation)
+				IdleHunt(unit)
+			end)
+		end
+
+		local entryPath = Utils.Random(HarkonnenPaths)
+		local units = Reinforcements.ReinforceWithTransport(harkonnen, "carryall.reinforce", HarkonnenReinforcements[Difficulty][wave], entryPath, { entryPath[1] })[2]
+		Utils.Do(units, function(unit)
+			unit.AttackMove(HarkonnenAttackLocation)
+			IdleHunt(unit)
+		end)
+
+		if wave < HarkonnenAttackWaves[Difficulty] then
+			SendHarkonnen()
+		end
+	end)
+end
+
+mercWave = 0
+SendMercenaries = function()
+	Trigger.AfterDelay(MercenaryAttackDelay[Difficulty], function()
+		mercWave = mercWave + 1
+
+		Media.DisplayMessage("Incoming hostile Mercenary force detected.", "Mentat")
+
+		local units = Reinforcements.Reinforce(mercenary, MercenaryReinforcements[Difficulty][mercWave], MercenarySpawn)
+		Utils.Do(units, function(unit)
+			unit.AttackMove(HarkonnenAttackLocation)
```

Comment:
Mercenaries attack the starport base first, if there is still a base, in original game, not main base.